index.js

```diff
@@ -90,100 +106,102 @@ exports.generate = function (opts, cb) {
+        }).reduce(function (total, contents) {
+            return total + os.EOL + contents;
 
-                // create path relative to opts.base
-                var relativeToBase = path.relative(path.resolve(opts.base), path.resolve(path.join(dir, filePath)));
+        // write contents to tmp file
+        }, '').then(function (css) {
+            var csspath = tempfile('.css');
+            return fs.writeFileAsync(csspath,css).then(function () {
+                return csspath;
+            });
 
-                // prepend / to make it absolute
-                return normalizePath(match.replace(filePath, path.join('/', relativeToBase)));
+        // let penthouseAsync do the rest
+        }).then(function (csspath) {
+            return penthouseAsync({
+                url: normalizePath(opts.url),
+                css: csspath,
+                // What viewports do you care about?
+                width: dimensions.width,   // viewport width
+                height: dimensions.height  // viewport height
             });
-        });
-
-    // combine all css files to one bid stylesheet
-    }).reduce(function (total, contents) {
-        return total + os.EOL + contents;
-
-    // write contents to tmp file
-    }, '').then(function (css) {
-        var csspath = tempfile('.css');
-        return fs.writeFileAsync(csspath,css).then(function () {
-            return csspath;
-        });
-
-    // let penthouseAsync do the rest
-    }).then(function (csspath) {
-        return penthouseAsync({
-            url: normalizePath(opts.url),
-            css: csspath,
-            // What viewports do you care about?
-            width: opts.width,   // viewport width
-            height: opts.height  // viewport height
-        });
-
-    // Penthouse callback
-    }).then(function (criticalCSS) {
-        if (opts.minify === true) {
-            criticalCSS = new CleanCSS().minify(criticalCSS);
-        }
 
-        if (opts.dest) {
-            // Write critical-path CSS
-            return fs.writeFileAsync(path.join(opts.base, opts.dest), criticalCSS).then(function () {
+        // Penthouse callback
+        }).then(function (criticalCSS) {
+            if (opts.minify === true) {
+                criticalCSS = new CleanCSS().minify(criticalCSS);
+            }
+
+            if (opts.dest) {
+                // Write critical-path CSS
+                return fs.writeFileAsync(path.join(opts.base, opts.dest), criticalCSS).then(function () {
+                    return criticalCSS;
+                });
+            } else {
                 return criticalCSS;
-            });
-        } else {
-            return criticalCSS;
-        }
+            }
 
-    // return err on error
-    }).then(function (criticalCSS) {
-        cb(null, criticalCSS.toString());
-    }).catch(function (err) {
+        // return err on error
+        })
+    })
+    .then(function (criticalCSS) {
+        cb(null, combineCss(criticalCSS));
```

Comment:
Great catch @bezoerb it very much should, or rather the write to disk should happen at this final step.

I will update the logic in this commit to do just that.